src/airflow_tools/providers/filesystem/operators/filesystem.py

```diff
@@ -106,3 +117,54 @@ def execute(self, context):
                 df.to_parquet(index=False, engine='pyarrow'), full_file_path
             )
             self.files.append(full_file_path)
+
+
+class FilesystemDeleteOperator(BaseOperator):
+    template_fields = ('filesystem_path',)
+
+    def __init__(self, filesystem_conn_id: str, filesystem_path: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.filesystem_conn_id = filesystem_conn_id
+        self.filesystem_path = filesystem_path
+
+    def execute(self, context: 'Context'):
+        filesystem_protocol = FilesystemFactory.get_data_lake_filesystem(
+            connection=BaseHook.get_connection(self.filesystem_conn_id),
+        )
+        filesystem_protocol.delete_prefix(self.filesystem_path)
+
+
+
+class FilesystemCheckOperator(BaseOperator):
+    template_fields = ('filesystem_path',)
+
+    def __init__(
+        self,
+        filesystem_conn_id: str,
+        filesystem_path: str,
+        check_specific_filename: None | str = None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.filesystem_conn_id = filesystem_conn_id
+        self.filesystem_path = filesystem_path
+        self.check_specific_filename = check_specific_filename
+
+    def execute(self, context: 'Context'):
+        filesystem_protocol = FilesystemFactory.get_data_lake_filesystem(
+            connection=BaseHook.get_connection(self.filesystem_conn_id),
+        )
+
+        logger.info(f'Checking {self.filesystem_path}')
+        prefix_flag = filesystem_protocol.check_prefix(self.filesystem_path)
+        
+        logger.info(f'Prefix flag: {prefix_flag}')
+        if self.check_specific_filename:
```

Comment:
Can you explain this logic? I don't fully understand it

Reply:
We have 2 use cases in this operator. One is to check if a prefix, exists. That would be up until the conditional

After that, we have the other use case, to check if we also have inside that prefix, a certain filename. This would allow us to check for __SUCCESS__ like files.

The function first checks if the prefix exists. If no specific file is requested, it returns the result of the checking. If another file is requested, it checks if the extra file exists. In this case, we only should return True, if both the specific file and the prefix both exists.